server/statistics/store.go

```diff
@@ -65,7 +65,24 @@ func (s *StoresStats) GetRollingStoreStats(storeID uint64) *RollingStoreStats {
 func (s *StoresStats) Observe(storeID uint64, stats *pdpb.StoreStats) {
 	s.RLock()
 	defer s.RUnlock()
-	s.rollingStoresStats[storeID].Observe(stats)
+	store, ok := s.rollingStoresStats[storeID]
+	if !ok {
+		store = newRollingStoreStats()
+		s.rollingStoresStats[storeID] = store
+	}
+	store.Observe(stats)
+}
+
+// Set sets the store statistics (for test).
+func (s *StoresStats) Set(storeID uint64, stats *pdpb.StoreStats) {
+	s.RLock()
```

Comment:
Lock()? `Observe` have the same  problem.  and Why not use Observe directly?

Reply:
`Observe`  `Add`s the new data into the `MovingAvg`, which will influenced by history. This one `Set`s the `MovingAvg` to the new data.